typings/vscode-mssql.d.ts

```diff
@@ -77,13 +77,13 @@ declare module 'vscode-mssql' {
 		getDatabaseNameFromTreeNode(node: ITreeNodeInfo): string;
 
 		/**
-		 * Get the connection string for the provided connection Uri
-		 * @param connectionUri The URI of the connection to get the connection string for.
-		 * @param includePassword Whether the Password is included in the connection string. Default is false.
-		 * @param includeApplicationName Whether the Application Name is included in the connection string. Default is true
-		 * @returns connection string
+		 * Get the connection string for the provided connection Uri or connection details.
+		 * @param connectionUriOrDetails Either the connection Uri for the connection or the connection details for the connection is required.
+		 * @param includePassword (optional) if password should be included in connection string.
+		 * @param includeApplicationName (optional) if application name should be included in connection string.
+		 * @returns connection string for the connection
 		 */
-		getConnectionString(connectionUri: String, includePassword?: boolean, includeApplicationName?: boolean): Promise<string>;
+		getConnectionString(connectionUriOrDetails: string | { [name: string]: any }, includePassword?: boolean, includeApplicationName?: boolean): Promise<string>;
```

Comment:
Would suggest moving the type definition into here if we're going to do this.

(we could also just not add it to the extension API for now. It isn't needed AFAIK)

Reply:
I tried adding ConnectionContracts.ConnectionDetails but was having trouble importing that into this file. Also, this is needed to be added so sql-database-projects can utilize it for disconnected scenarios.

Reply:
What "problems"? Feel free to reach out if you're having issues with stuff like this